great_expectations/datasource/fluent/interfaces.py

```diff
@@ -994,7 +1000,7 @@ def __init__(  # noqa: PLR0913
     def _create_id(self) -> str:
         options_list = []
         for key, value in self.batch_request.options.items():
-            if key != "path":
+            if key not in ("path", "dataframe"):
```

Comment:
We don't want the `dataframe` to be part of the Batch id. We should probably have this configurable at the asset/batch definition level since it seems strange for the top level batch to know specifics about different assets/batch definitions that produce them, but I am following the pattern here and we can refactor later.